packages/provider-elasticsearch/src/example-types/filters/tagsQuery.js

```diff
@@ -80,28 +81,66 @@ let filter = ({ tags, join, field, exact }) => ({
   },
 })
 
-let buildResultQuery = (node, children = {}, groupsKey = 'tags') => ({
-  aggs: {
-    [groupsKey]: {
-      filters: {
-        filters: F.arrayToObject(
-          _.get('word'),
-          (tag) => filter({ ...node, tags: [tag] }),
-          node.tags
-        ),
+let buildResultQuery = (
+  node,
+  children = {},
+  groupsKey = 'tags',
+  keywordGenerations = []
+) => {
+  return {
+    aggs: {
+      [groupsKey]: {
+        filters: {
+          filters: F.arrayToObject(
+            _.get('word'),
+            (tag) => filter({ ...node, tags: [tag] }),
+            node.tags
+          ),
+        },
+        ...children,
       },
-      ...children,
+      ...(keywordGenerations.length > 0 && {
+        keywordGenerations: {
+          filters: {
+            filters: F.compactObject(
+              F.arrayToObject(
+                _.identity,
+                (word) => filter({ ...node, tags: [{ word: `${word}` }] }),
+                keywordGenerations
+              )
+            ),
+          },
+          aggs: {
+            keyword_generation_sort: {
+              bucket_sort: {
+                sort: [{ _count: { order: 'desc' } }],
+              },
+            },
+          },
+        },
+      }),
     },
-  },
-})
+  }
+}
 
-let result = async (node, search) => {
-  let aggs = buildResultQuery(node)
+let result = (generateKeywords) => async (node, search) => {
+  let keywords =
+    node.generateKeywords === true
+      ? await generateKeywords(generationTagInputs(node.tags))
+      : []
+  let aggs = node.generateKeywords
+    ? buildResultQuery(node, {}, 'tags', keywords)
+    : buildResultQuery(node)
```

Comment:
it still works in the pivot scenario because it uses buildResultQuery to build it up from what I remember. I can take a look again but all of those tests are passing right now for pivot